### PR TITLE
HS-175 Convert REST API server to BFF component with GraphQL support

### PR DESCRIPTION
**Pull request by yangli2014**
## JIRA Link
https://issues.opennms.org/browse/HS-175
https://issues.opennms.org/browse/HS-209
https://issues.opennms.org/browse/HS-210
## Checklist
* [ ] JIRA has been updated to 'in review'.
* [ ] Unit tests have been added as part of the PR.
* [ ] Manual tests have been completed, and notes added to JIRA.
* [ ] Details have been provided to docs team (if necessary)
* [ ] Screenshot(s) included (applicable to changes with a visual impact).

## Details of the change

- The Alarms and Events Rest API endpoints still exist. The API doc can be accessed at http://localhost:9090/swagger-ui/index.html
- The GraphQL schema and playground can be accessed at http://localhost:9090/gui
- The Nginx API gateway is configured to point to the API server.

### CODE REVIEW

**Comment by yangli2014** [comment]
> 

As we discussed I will address it in HS-210

File: platform/events/api/src/main/java/org/opennms/horizon/events/xml/DateTimeAdapter.java [hunk 1]
```diff
@@ -45,6 +45,12 @@ public String marshal(final Date date) throws Exception {
     /** {@inheritDoc} */
     @Override
     public Date unmarshal(final String string) throws Exception {
+        try{
+            Long time = Long.valueOf(string);
+            return new Date(time);
+        } catch (NumberFormatException e){
+
```
Comment (geraldhumphries) [inline platform/events/api/src/main/java/org/opennms/horizon/events/xml/DateTimeAdapter.java:52]: We should at least log something on exception here.

**Review by geraldhumphries — CHANGES_REQUESTED** [review]
Feels really good seeing a PR that removes so much code :)

File: platform/events/api/src/main/java/org/opennms/horizon/events/xml/DateTimeAdapter.java (hunk 1 above)
Comment (yangli2014) [reply platform/events/api/src/main/java/org/opennms/horizon/events/xml/DateTimeAdapter.java:52]: Sure. The code here is handle the date in timestamp or ISO string format.

Comment (yangli2014) [reply platform/events/api/src/main/java/org/opennms/horizon/events/xml/DateTimeAdapter.java]: After I cleaned up, this change is not needed anymore.

**Review by geraldhumphries — APPROVED** [review]

**Review by mfrazier63 — APPROVED** [review]
Looks much better!

**Pull request merged**